packages/cli/src/commands/multi-chain/deploy.ts

```diff
@@ -0,0 +1,295 @@
+// Copyright 2020-2024 SubQuery Pte Ltd authors & contributors
+// SPDX-License-Identifier: GPL-3.0
+
+import path from 'path';
+import {Command, Flags} from '@oclif/core';
+import chalk from 'chalk';
+import cli from 'cli-ux';
+import inquirer from 'inquirer';
+import {BASE_PROJECT_URL, DEFAULT_DEPLOYMENT_TYPE, ROOT_API_URL_PROD} from '../../constants';
+import YAML from 'yaml'
+import {
+  createDeployment,
+  dictionaryEndpoints,
+  imageVersions,
+  ipfsCID_validate,
+  processEndpoints,
+  projectsInfo,
+  splitEndpoints, splitMultichainDataFields,
+  updateDeployment,
+} from '../../controller/deploy-controller';
+import {
+  IndexerAdvancedOpts,
+  multichainDataFieldType,
+  QueryAdvancedOpts,
+  V3DeploymentIndexerType
+} from '../../types';
+import {addV, checkToken, promptWithDefaultValues, resolveToAbsolutePath, valueOrPrompt} from '../../utils';
+import MultiChainAdd from "./add";
+import Build from "../build";
+import {getMultichainManifestPath, getProjectRootAndManifest} from "@subql/common";
+import fs, {existsSync, readFileSync} from "fs";
+import {uploadToIpfs} from "../../controller/publish-controller";
+
+const ACCESS_TOKEN_PATH = path.resolve(process.env.HOME, '.subql/SUBQL_ACCESS_TOKEN');
+
+export default class MultiChainDeploy extends Command {
```

Comment:
There is a lot of code here that is duplicated with the normal deploy function. It would be good to reuse a much as possible, particularly with the flags.

Reply:
reorganized